src/widgets/argocd/component.jsx

```diff
@@ -0,0 +1,52 @@
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+export default function Component({ service }) {
+  const { widget } = service;
+
+  // Limits fields to available statuses
+  const validFields = ["apps", "synced", "outOfSync", "healthy", "progressing", "degraded", "suspended", "missing"];
+  widget.fields = widget.fields?.filter((field) => validFields.includes(field));
+
+  // Limits max number of displayed fields
```

Comment:
```suggestion
  if (!widget.fields) {
    widget.fields = ["apps", "synced", "outOfSync", "healthy"];
  }

```

1. we dont do this kind of field validation here
2. At the moment the widget doesn't work if the user doesn't supply any fields, which is pretty common

Reply:
Cool, good to know